### PR TITLE
feat: [PAYM-1572] Update react-number-format, introduce PatternField

### PR DESCRIPTION
**Pull request by lapapooh**
**PR description:**
Because of business card we had to upgrade `react-number-format` from v4 to v5. In v5, on top of new customization helpers, NumberFormat was divided into 2 separate components: NumericFormat, PatternFormat. Hence, we are introducing new PatternField component.
[Migration guide](https://s-yadav.github.io/react-number-format/docs/migration)

**Implemented:**

**Checklist:**

- [x] I ran this code locally
- [ ] I wrote the necessary tests
- [x] My code follows the [style guidelines](http://bit.ly/sd-web-style-guide)
- [x] I followed the [instructions](http://bit.ly/sd-web-pr) to create a pull request

**JIRA card:**
https://superdispatch.atlassian.net/browse/PAYM-1572

**Should know about:**

### CODE REVIEW

File: packages/ui/playroom.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export {
   Inline,
   NumberField,
   OverflowText,
+  PatternField,
```
Comment (doniyor2109) [inline packages/ui/playroom.ts:21]: Lets add formik integration as well

File: packages/ui/src/pattern-field/PatternField.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    >,
+    SafePatternFormatProps {}
+
+export function PatternField({
+  value,
+  inputRef,
+  inputMode = 'decimal',
+  valueIsNumericString = true,
+  ...props
+}: PatternFieldProps): JSX.Element {
```
Comment (doniyor2109) [inline packages/ui/src/pattern-field/PatternField.tsx:34]: Lets use forwardRef here

**Review by timurhamitov — APPROVED** [review]

File: packages/forms/src/text/FormikPatternField.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  return error || undefined;
+}
+
+export interface FormikPatternFieldProps extends PatternFieldProps {
+  name: string;
+  validate?: FieldValidator;
+  formatError?: (error: string) => ReactNode;
+}
+
+export const FormikPatternField: ForwardRefExoticComponent<FormikPatternFieldProps> =
```
Comment (doniyor2109) [inline packages/forms/src/text/FormikPatternField.tsx:15]: Lets export this component from index.ts

File: packages/ui/src/pattern-field/__test__/PatternField.spec.tsx
Comment (doniyor2109) [inline packages/ui/src/pattern-field/__test__/PatternField.spec.tsx:1]: It is better to write functional tests instead of style tests

**Review by doniyor2109 — APPROVED** [review]

**Pull request merged**